### PR TITLE
Recipe for fslpy

### PR DESCRIPTION
**Pull request by pauldmccarthy**
Recipe for fslpy, the FSL python library:
  - http://git.fmrib.ox.ac.uk/fsl/fslpy
  - https://pypi.org/project/fslpy/

### CODE REVIEW

**Comment by pauldmccarthy** [comment]
Hi @conda-forge/staged-recipes, this recipe is ready for review - thanks!

File: recipes/fslpy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
```
Comment (marcelotrevisani) [inline recipes/fslpy/meta.yaml]: Could you please change this part to
```
script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
```

File: recipes/fslpy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+requirements:
+  build:
+    - python
+    - pip
+    - sphinx 1.6.*
+    - sphinx_rtd_theme 0.*
+    - mock 2.*
+    - coverage 4.*
+    - pytest 3.*
```
Comment (marcelotrevisani) [inline recipes/fslpy/meta.yaml]: do you need pytest just to test your project or you really need at build time?

Comment (marcelotrevisani) [reply recipes/fslpy/meta.yaml]: and also mock, pytest-coverage, coverage

File: recipes/fslpy/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    - pytest-cov 2.*
+  run:
+    - python
+    - deprecation >=1.*,<=2.*
+    - numpy 1.*
```
Comment (marcelotrevisani) [inline recipes/fslpy/meta.yaml]: Could you please change this to
```
- {{ pin_compatible('numpy') }}
```

File: recipes/fslpy/meta.yaml (hunk 2 above)
Comment (pauldmccarthy) [reply recipes/fslpy/meta.yaml]: No, I copied as-is from the setup.py (where I've bundled build+test deps together). Will change.

File: recipes/fslpy/meta.yaml (hunk 1 above)
Comment (pauldmccarthy) [reply recipes/fslpy/meta.yaml]: @marcelotrevisani  All builds now failing with `Undefined Jinja2 variables remain (['PYTHON'])`. Any ideas?

Comment (pauldmccarthy) [reply recipes/fslpy/meta.yaml]: Any problem with me reverting this change? It was working fine before.

**Comment by pauldmccarthy** [comment]
@marcelotrevisani, ready for another review i think! The build deps do need to be present, as they are listed in the `setup_requires` section of `setup.py`.

**Comment by pauldmccarthy** [comment]
Hi @conda-forge/staged-recipes @marcelotrevisani, any chance of having this looked at again? Thanks!

File: recipes/fslpy/meta.yaml (hunk 2 above)
Comment (marcelotrevisani) [reply recipes/fslpy/meta.yaml]: Ah ok, I just saw your ``setup.py`` that is using the ``requirements-dev.txt`` to the setup process. Would be good to change that, but as your ``setup.py`` requires that it has to be here indeed.  :)

File: recipes/fslpy/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    - atlasquery    = fsl.scripts.atlasq:atlasquery_emulation
+    - extract_noise = fsl.scripts.extract_noise:main
+
+requirements:
+  build:
+    - python
+    - pip
+  host:
+    - python
+    - pip
```
Comment (marcelotrevisani) [inline recipes/fslpy/meta.yaml]: Please remove ``pip`` from here

**Comment by pauldmccarthy** [comment]
@marcelotrevisani no problemo, hope you enjoyed the vacation, and thanks for getting back to me!

**Comment by marcelotrevisani** [comment]
Thanks! Looks good to me, I'm going to merge this!

Thanks to put this in ``conda-forge``

File: recipes/fslpy/meta.yaml (hunk 4 above)
Comment (isuruf) [reply recipes/fslpy/meta.yaml]: Why?

**Pull request merged**